src/StreamJsonRpc/Reflection/RpcTargetInfo.cs

```diff
@@ -266,6 +239,27 @@ internal bool TryGetTargetMethod(JsonRpcRequest request, [NotNullWhen(true)] out
         return revert;
     }
 
+    /// <summary>
+    /// Adds a new RPC interface to an existing target registering additional RPC methods.
+    /// </summary>
+    /// <param name="exposingMembersOn">The interface type whose members define the RPC accessible members of the <paramref name="target"/> object.</param>
+    /// <param name="target">Target to invoke when incoming messages are received.</param>
+    /// <param name="options">A set of customizations for how the target object is registered. If <c>null</c>, default options will be used.</param>
+    /// <param name="revertAddLocalRpcTarget">An optional object that may be disposed of to revert the addition of the target object. This object is returned by an earlier call to <see cref="AddRpcInterfaceToTarget(Type, object, JsonRpcTargetOptions?, IDisposable?)"/>.</param>
+    internal void AddRpcInterfaceToTarget(Type exposingMembersOn, object target, JsonRpcTargetOptions? options, IDisposable? revertAddLocalRpcTarget)
+    {
+        RevertAddLocalRpcTarget? revert = revertAddLocalRpcTarget as RevertAddLocalRpcTarget;
+        if (revertAddLocalRpcTarget is not null && revert is null)
+        {
+            throw new ArgumentException(nameof(revertAddLocalRpcTarget));
+        }
+
```

Comment:
`RevertAddLocalRpcTarget` is a private nested class, so I can't expose it from an internal method. I can make the `RevertAddLocalRpcTarget` `internal` though if we want to avoid this